src/core/web_api/p2_api.js

```diff
@@ -82,6 +60,50 @@ class Auth0APIClient {
       this.authOpt.sso = opts.sso;
     }
   }
+  getTelemetryInfo(telemetryOverride) {
+    let telemetry;
+    const { auth0Client } = qs.parse(window.location.search.substr(1));
+    let ulpTelemetry = auth0Client && JSON.parse(atob(auth0Client));
+    if (this.isUniversalLogin && ulpTelemetry) {
+      telemetry = {
+        ...ulpTelemetry,
+        env: {
+          ...ulpTelemetry.env,
+          ['lock.js-ulp']: getVersion(),
+          ['auth0-js-ulp']: auth0.version.raw
+        }
+      };
+    }
+    if (this.isUniversalLogin && !ulpTelemetry) {
+      telemetry = {
+        name: 'lock.js-ulp',
+        version: getVersion(),
+        env: {
+          ['auth0-js-ulp']: auth0.version.raw
+        }
+      };
+    }
+    if (!this.isUniversalLogin && telemetryOverride) {
+      telemetry = {
+        ...telemetryOverride,
```

Comment:
what if this already has an `env` property? how would that behave? I would like it to me merged

Reply:
see two lines down